### PR TITLE
Python 3.6 support

### LINKED ISSUE #1650: Make Blaze Python 3.6 compatible
Blaze has a fixed dependency on psutil 4.0.0 that limits blaze to python 3.5. Looking at how psutil is used should be no problem to bump it's version. Not sure what other packages may need to be updated.

### PR DESCRIPTION
**Pull request by postelrich**
This PR removes the pinned version of psutil. That is the only thing blocking 3.6 versions from being built.

Resolves #1650

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,19 +15,24 @@ matrix:
...
       env: SPARK_VERSION=1.5
+    - python: 3.6
     - python: 2.7
       env: PANDAS_VERSION="git+https://github.com/pydata/pandas" SPARK_VERSION=1.5
     - python: 3.4
       env: PANDAS_VERSION="git+https://github.com/pydata/pandas" SPARK_VERSION=1.5
     - python: 3.5
       env: PANDAS_VERSION="git+https://github.com/pydata/pandas" SPARK_VERSION=1.5
+    - python: 3.6
+      env: PANDAS_VERSION="git+https://github.com/pydata/pandas"
```
Comment (mcg1969) [inline .travis.yml]: Is there a reason you did not duplicate the `SPARK_VERSION` env variable here?

File: .travis.yml [hunk 2]
```diff
@@ -49,13 +54,13 @@ install:
...
   - conda create -n odo numpy=1.11.2
-  - conda create -n blaze -c blaze python=$TRAVIS_PYTHON_VERSION numpy=1.11.2 numba==0.29.0
+  - conda create -n blaze -c blaze python=$TRAVIS_PYTHON_VERSION numpy=1.11.2 numba==0.34.0
   - if [ -n "$PANDAS_VERSION" ]; then pip install cython==0.24.1; pip install $PANDAS_VERSION; else conda install pandas=0.19.0; fi
   - source activate blaze
   # update setuptools and pip
   - conda update setuptools pip
 
-  - conda install spark=$SPARK_VERSION -c blaze -c anaconda-cluster
+  - if [ -n "$SPARK_VERSION" ]; then conda install spark=$SPARK_VERSION -c blaze -c anaconda-cluster; else conda install pyspark -c conda-forge; fi
```
Comment (mcg1969) [inline .travis.yml]: Ah, I think this helps explain it.

**Comment by lia-simeone** [comment]
@mcg1969 Any hope of getting this PR merged?

**Comment by lia-simeone** [comment]
@mcg1969 @postelrich any update here? It's been over a year since the PR opened 😂

**Comment by postelrich** [comment]
@lia-simeone closing this in favor of https://github.com/blaze/blaze/pull/1671 which I've also had a lack of time to work on.

**Pull request closed without merging**